aspnet/mvc/views/razor.rst

```diff
@@ -1,8 +1,689 @@
+  }
+  <!-- HTML comment -->
+
+Is rendered by the server as:
+
+.. code-block:: none
+
+  <!-- HTML comment -->
+
+Razor comments are removed by the server before the page is rendered. Razor uses ``@*  *@`` to delimit comments. The following code is commented out, so the server will not render any markup:
+
+.. code-block:: none
+
+    @*
+    @{
+        /* C# comment. */
+        // Another C# comment.
+    }
+    <!-- HTML comment -->
+   *@
+
+.. _razor-directives-label:
+
+Directives
+-----------
+Razor directives are represented by implicit expressions with reserved keywords following the ``@`` symbol. A directive will typically change the way a page is parsed or enable different functionality within your Razor page.
+
+Understanding how Razor generates code for a view will make it easier to understand how directives work. A Razor page is used to generate a C# file. For example, this Razor page:
+
+.. literalinclude:: razor/sample/Views/Home/Contact8.cshtml
+  :language: html
+
+Generate a class similar to the following:
+
+.. code-block:: c#
+
+  public class _Views_Something_cshtml : RazorPage<dynamic>
+  {
+      public override async Task ExecuteAsync()
+      {
+          var output = "Hello World";
+
+          WriteLiteral("/r/n<div>Output: ");
+          Write(output);
+          WriteLiteral("</div>");
+      }
+  }
+
+:ref:`razor-customcompilationservice-label` explains how to view this generated class.
+
+``@using``
+^^^^^^^^^^^^^
+
+The ``@using`` directive will add the c# ``using`` directive to the generated razor page:
+
+.. literalinclude:: razor/sample/Views/Home/Contact9.cshtml
+  :language: html
+
+``@model``
+^^^^^^^^^^^^
+
+The ``@model`` directive allows you to specify the type of the model passed to your Razor page. It uses the following syntax:
+
+.. code-block:: none
+
+  @model TypeNameOfModel<TModel>
+
+For example, if you create an ASP.NET Core MVC app with individual user accounts, the *Views/Account/Login.cshtml* Razor view contains the follow model declaration:
+
+.. code-block:: c#
+
+  @model LoginViewModel
+
+In the class example in :ref:`razor-directives-label`, the class generated inherits from ``RazorPage<dynamic>``. By adding an ``@model`` you control what’s inherited. For example
+
+.. code-block:: c#
+
+  @model LoginViewModel
+
+Generates the following class
+
+.. code-block:: c#
+
+ public class _Views_Account_Login_cshtml : RazorPage<LoginViewModel>
+
+Razor pages expose a ``Model`` property for accessing the model passed to the page.
+
+.. code-block:: html
+
+  <div>The Login Email: @Model.Email</div>
+
+The ``@model`` directive specified the type of this property (by specifying the ``T`` in ``RazorPage<T>`` that the generated class for your page derives from). If you don't specify the ``@model`` directive the ``Model`` property will be of type ``dynamic``. The value of the model is passed from the controller to the view. See :ref:`strongly-typed-models-keyword-label` for more information.
```

Comment:
> :ref:`strongly-typed-models-keyword-label`

This ref still isn't resolving and I don't see the label defined anywhere. Did you mean to create it in the views overview doc?